src/tabs.ts

```diff
@@ -0,0 +1,258 @@
+export class AalamTabs extends LitElement {
+    DEFAULT_VALUES:{ [key:string]: any } = {
+        animationDur: 100,
+        activecls: 'tab-active',
+        fashion: 'xs:row',
+        colsize: 'title:30%;body:70%',
+        internal_fashion: 'row',
+        fashion_style: {xs:'row'},
+        column_size:{title:'30%', body:'70%'}
+    }
+
+    @property({type:String, attribute:true})
+    animation = '';
+
+    @property({type:Number, attribute:true})
+    animationDur = this.DEFAULT_VALUES.animationDur;
+
+    @property({type:String, attribute:true})
+    activecls = this.DEFAULT_VALUES.activecls;
+
+    @property({type:String, attribute:true})
+    fashion = this.DEFAULT_VALUES.fashion;
+
+    @property({type:String, attribute:true})
+    colsize = this.DEFAULT_VALUES.colsize;
+
+    @state()
+    _internal_fashion:string = this.DEFAULT_VALUES.internal_fashion;
+
+    private _prev_ix:number;
+    private _resizeListener = this._resizeEvent.bind(this);
+    private _animation_styles:{[key:string]:string} = {};
+    private _column_size:{[key:string]:string} =
+                this.DEFAULT_VALUES.column_size;
+    private _fashion_style:Array<ResponsiveVal> =
+                this.DEFAULT_VALUES.fashion_style;
+
+    constructor() {
+        super();
+    }
+    override attributeChangedCallback(
+                name:string, old_val:string, new_val:string) {
+        super.attributeChangedCallback(name, old_val, new_val);
+        if(name == 'animation') {
+            if(animate_defs[new_val])
+                this._animation_styles = {open: new_val, close: new_val}
+            else {
+                this._animation_styles = parseAttrVal(new_val);
+                this._animation_styles = Object.assign({}, {
+                    open:this._animation_styles?.show,
+                    close:this._animation_styles?.hide});
+            }
+        } else if(name == 'fashion') {
+            this._fashion_style = getResponsiveValues(
+                new_val, this.DEFAULT_VALUES.fashion);
+            let val = this._checkFashion();
+            this._changeFashionStyle(val);
+        } else if(name == "colsize")
+            this._column_size = parseAttrVal(new_val);
+    }
+    override render() {
+        if(this._internal_fashion == 'accordion') {
+            return html`
+<div id='accordion' @click=${this._titleClicked}>
+    <slot name='acc'></slot>
+</div>
+            `
+        } else {
+            let fashionColumn:any = {
+                display: 'grid',
+                'grid-template-columns': `
+                    ${this._column_size.title} ${this._column_size.body}`
+            }
+            return html`
+${this._animateStyles()}
+<div id="parent" class="parentDiv"
+    style=${this._internal_fashion=='column'?
+            styleMap(fashionColumn):''}>
+    <div id="title-tab" style=
+        "display:flex;
+         flex-direction:${this._internal_fashion=='column'?
+            'column':'row'}">
```

Comment:
It will better we can just assign flex-direction to ${this._internal_fashion} as it will be either "row" or "column" in itself.